app/app.go

```diff
@@ -0,0 +1,70 @@
+package app
+
+import (
+	"database/sql"
+
+	"golang.org/x/crypto/bcrypt"
+)
+
+//nema pojma za db i web
+
+type Application struct {
+	UserService *UserService
+}
+type User struct {
+	FirstName string
+	LastName  string
+	Email     string
+	Username  string
+	Password  string
+}
+
+type UserService struct {
+	db *sql.DB
+}
+
+func NewUserService(db *sql.DB) *UserService {
+	return &UserService{
+		db,
+	}
+}
+
+type CreateUserRequest struct {
+	FirstName string
+	LastName  string
+	Email     string
+	Username  string
+	Password  string
+}
+
+func HashPassword(password string) (string, error) {
+	bytes, err := bcrypt.GenerateFromPassword([]byte(password), 14)
+	return string(bytes), err
+}
+
+func (s *UserService) Create(req CreateUserRequest) (*User, error) {
+
+	hashedPassword, err := HashPassword(req.Password)
+	if err != nil {
+		return nil, err
+	}
+	_, err = s.db.Exec("INSERT INTO users (first_name, last_name, email, username, password) VALUES ($1, $2, $3, $4, $5)",
```

Comment:
Running these queries should be the responsibility of the `UserRepository` implementation in the `postgres` package. So you should create a `UserRepository` struct in the `postgres` package and have it receive the `sql.DB` instance. The `UserService` would then receive the `UserRepository` in the constructor function and would call it here:
```go
func (s *UserService) Create(req CreateUserRequest) (*User, error) {
    // ...
    user, err := s.repository.Create(req)
    // ...
}
```

Pay attention that this would be the interface `UserRepository` and not the concrete implementation type `postgres.UserRepository`. This enables us to inject any implementation for a repository, not just postgres.